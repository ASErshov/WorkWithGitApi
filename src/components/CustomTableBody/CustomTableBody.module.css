.tableBody{
    border-left: solid 1px rgba(224, 224, 224, 1);;
    border-right: solid 1px rgba(224, 224, 224, 1);;
}