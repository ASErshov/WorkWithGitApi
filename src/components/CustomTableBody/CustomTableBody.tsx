import React from 'react'
import {TableRow, TableCell, TableBody } from '@material-ui/core'
import styles from './CustomTableBody.module.css'

export type TableBodyProps= {
    records: Array<Array<string|number>>
}

const CustomTableBody: React.FC<TableBodyProps> = ({records}):React.ReactElement => {
    return(
        <TableBody className={styles.tableBody}>
          { records.map((row, indexRow) => (
            <TableRow key={`rowindex_${indexRow}`}>
               {row.map((item, indexItem)=>(
               <TableCell align='left' key={`cellIndex_${indexItem}`}>
                   {typeof item === 'string' && item.split(':')[0]==='https' ? <a href={item}>{item}</a>: item}
               </TableCell>
               ))}
            </TableRow>
          ))}
        </TableBody>
    )
}

export default CustomTableBody