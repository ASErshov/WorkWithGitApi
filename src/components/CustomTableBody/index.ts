export { default } from './CustomTableBody'
export * from './CustomTableBody'