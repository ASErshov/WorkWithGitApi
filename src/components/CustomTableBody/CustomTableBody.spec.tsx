import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import CustomTableBody from './index'

describe('<CustomTableBody />', () => {
  const Component = (
    <CustomTableBody records={[[1,2,3],[4,5,6],[7,8,9]]}/>
  )

  it('Snapshot is correct', () => {
    const wrapper = renderer.create(Component).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('Passed props is valid', () => {
    const wrapper = mount(Component)

    expect(wrapper.props().records).toEqual([[1,2,3],[4,5,6],[7,8,9]])

    wrapper.unmount()
  })

})
