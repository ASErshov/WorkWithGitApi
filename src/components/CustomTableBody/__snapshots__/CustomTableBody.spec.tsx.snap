// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTableBody /> Snapshot is correct 1`] = `
<tbody
  className="MuiTableBody-root tableBody"
  role={null}
>
  <tr
    className="MuiTableRow-root"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      1
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      2
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      3
    </td>
  </tr>
  <tr
    className="MuiTableRow-root"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      4
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      5
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      6
    </td>
  </tr>
  <tr
    className="MuiTableRow-root"
    role={null}
  >
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      7
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      8
    </td>
    <td
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
    >
      9
    </td>
  </tr>
</tbody>
`;
